Tests/FuzzilliTests/ProgramBuilderTest.swift

```diff
@@ -251,6 +251,445 @@ class ProgramBuilderTests: XCTestCase {
             }
         }
     }
+
+    func testClassSplicing() {
+        let fuzzer = makeMockFuzzer()
+        let b = fuzzer.makeBuilder()
+        // We enable conservative mode to exercise the canMutate checks within the splice loop
+        b.mode = .conservative
+
+        var superclass = b.defineClass() { cls in
+            cls.defineConstructor(withParameters: [.integer]) { params in
+            }
+
+            cls.defineProperty("a")
+
+            cls.defineMethod("f", withSignature: [.float] => .string) { params in
+                b.doReturn(value: b.loadString("foobar"))
+            }
+        }
+
+        let _ = b.defineClass(withSuperclass: superclass) { cls in
+            cls.defineConstructor(withParameters: [.string]) { params in
+                let v3 = b.loadInt(0)
+                let v4 = b.loadInt(2)
+                let v5 = b.loadInt(1)
+                b.forLoop(v3, .lessThan, v4, .Add, v5) { _ in
+                    let v0 = b.loadInt(42)
+                    let v1 = b.createObject(with: ["foo": v0])
+                    b.callSuperConstructor(withArgs: [v1])
+                }
+            }
+
+            cls.defineProperty("b")
+
+            cls.defineMethod("g", withSignature: [.anything] => .unknown) { params in
+                b.definePlainFunction(withSignature: [] => .unknown) { _ in
+                }
+            }
+        }
+
+        let original = b.finalize()
+
+        superclass = b.defineClass() { cls in
+            cls.defineConstructor(withParameters: [.integer]) { params in
+            }
+        }
+        b.defineClass(withSuperclass: superclass) { cls in
+            cls.defineConstructor(withParameters: [.string]) { _ in
+                // Splicing at CallSuperConstructor
+                b.splice(from: original, at: original.code.lastInstruction.index - 5)
```

Comment:
Even with the comments, I still find it a bit tricky to verify that these splicing indices are correct... how about this:
Add a method
```swift
    /// Returns the index of the next instruction added to the program. This is equal to the current size of the program.
    public func indexOfNextInstruction() {
        return code.count
    }
```
To the ProgramBuilder, for example just below `dumpCurrentProgram`.
Then change these tests to be structured like this:
```swift
var splicePoint = -1
let b = fuzzer.makeBuilder()

// ...

        let v1 = b.createObject(with: ["foo": v0])
        splicePoint = b.indexOfNextInstruction()
        b.callSuperConstructor(withArgs: [v1])      // Splice here
     }

// ...

        b.splice(from: original, at: splicePoint)
```
That'll make it obvious where the splicing happens and will also guarantee that the test still works correctly if new instructions are added to the program.